extension/src/popup/helpers/soroban.ts

```diff
@@ -327,18 +327,90 @@ export function pickTransfers(invocationTree: InvocationTree) {
 export function getInvocationDetails(
   invocation: xdr.SorobanAuthorizedInvocation,
 ) {
-  return [
+  const invocations = [
     getInvocationArgs(invocation),
     ...invocation.subInvocations().map(getInvocationArgs),
   ];
+  return invocations.filter(isInvocationArg);
 }
 
-function getInvocationArgs(invocation: xdr.SorobanAuthorizedInvocation) {
-  const _invocation = invocation.function().contractFn();
-  const contractId = StrKey.encodeContract(
-    _invocation.contractAddress().contractId(),
-  );
-  const fnName = _invocation.functionName().toString();
-  const args = _invocation.args();
-  return { fnName, contractId, args };
+export interface FnArgsInvoke {
+  type: "invoke";
+  fnName: string;
+  contractId: string;
+  args: xdr.ScVal[];
+}
+
+export interface FnArgsCreateWasm {
+  type: "wasm";
+  salt: string;
+  hash: string;
+  address: string;
+}
+
+export interface FnArgsCreateSac {
+  type: "sac";
+  asset: string;
+}
+
+type InvocationArgs = FnArgsInvoke | FnArgsCreateWasm | FnArgsCreateSac;
+
+const isInvocationArg = (
+  invocation: InvocationArgs | undefined,
+): invocation is InvocationArgs => !!invocation;
+
+function getInvocationArgs(
```

Comment:
This fn has a good amount of overlap with 'buildInvocationTree', there should be the opportunity to combine these in the future but I didn't do it today in order to keep the change set as small as possible before the vote.